Caesars Cipher_MP_1101.py

```diff
@@ -1,224 +1,225 @@
+    '''
+    print('Loading word list from file...')
+    # inFile: file
+    in_file = open(file_name, 'r')
+    # line: string
+    line = in_file.readline()
+    # word_list: list of strings
+    word_list = line.split()
+    print('  ', len(word_list), 'words loaded.')
+    in_file.close()
+    return word_list
+
+### DO NOT MODIFY THIS FUNCTION ###
+def is_word(word_list, word):
+    '''
+    Determines if word is a valid word, ignoring
+    capitalization and punctuation
+
+    word_list (list): list of words in the dictionary.
+    word (string): a possible word.
+    
+    Returns: True if word is in word_list, False otherwise
+
+    Example:
+    >>> is_word(word_list, 'bat') returns
+    True
+    >>> is_word(word_list, 'asdf') returns
+    False
+    '''
+    word = word.lower()
+    word = word.strip(" !@#$%^&*()-_+={}[]|\:;'<>?,./\"")
+    return word in word_list
+
+### DO NOT MODIFY THIS FUNCTION ###
+def get_story_string():
+    """
+    Returns: a joke in encrypted text.
+    """
+    f = open("story.txt", "r")
+    story = str(f.read())
+    f.close()
+    return story
+
+WORDLIST_FILENAME = 'words.txt'
+
+class Message(object):
+    ### DO NOT MODIFY THIS METHOD ###
+    def __init__(self, text):
+        '''
+        Initializes a Message object
+                
+        text (string): the message's text
+
+        a Message object has two attributes:
+            self.message_text (string, determined by input text)
+            self.valid_words (list, determined using helper function load_words
+        '''
+        self.message_text = text
+        self.valid_words = load_words(WORDLIST_FILENAME)
+
+    ### DO NOT MODIFY THIS METHOD ###
+    def get_message_text(self):
+        '''
+        Used to safely access self.message_text outside of the class
+        
+        Returns: self.message_text
+        '''
+        return self.message_text
+
+    ### DO NOT MODIFY THIS METHOD ###
+    def get_valid_words(self):
+        '''
+        Used to safely access a copy of self.valid_words outside of the class
+        
+        Returns: a COPY of self.valid_words
+        '''
+        return self.valid_words[:]
+        
+    def build_shift_dict(self, shift):
+        '''
+        Creates a dictionary that can be used to apply a cipher to a letter.
+        The dictionary maps every uppercase and lowercase letter to a
+        character shifted down the alphabet by the input shift. The dictionary
+        should have 52 keys of all the uppercase letters and all the lowercase
+        letters only.        
+        
+        shift (integer): the amount by which to shift every letter of the 
+        alphabet. 0 <= shift < 26
+
+        Returns: a dictionary mapping a letter (string) to 
+                 another letter (string). 
+        '''
+        alphabet = string.ascii_lowercase + string.ascii_uppercase
+        return {current: alphabet[(alphabet.index(current) + shift) % 26] for current in alphabet}
```

Comment:
Exception has occurred: TypeError
unsupported operand type(s) for +: 'int' and 'str'
 Cipher_MP_1101.py", line 106, in <dictcomp>
    return {current: alphabet[(alphabet.index(current) + shift) % 26] for current in alphabet}

@General-MP